### PR TITLE
Updated path completion for long path names to display the completed …

### PR DESCRIPTION
**Pull request by emjobson**
…portion rather than the prefix. Will now elide for long file names as well as long object names.

https://github.com/ipython/ipython/issues/11441

This is my first PR. Please let me know if you have any feedback!

### CODE REVIEW

File: IPython/terminal/ptutils.py [hunk 1]
```diff
@@ -36,12 +39,16 @@ def _elide(string, *, min_elide=30):
...
         return string
 
-    parts = string.split('.')
+    object_parts = string.split('.')
+    file_parts = string.split('/')
 
-    if len(parts) <= 3:
-        return string
+    if len(object_parts) > 3:
+        return '{}.{}\N{HORIZONTAL ELLIPSIS}{}.{}'.format(parts[0], parts[1][0], parts[-2][-1], parts[-1])
```
Comment (Carreau) [inline IPython/terminal/ptutils.py]: you've kept `parts` here, which does not exist any more. Did you mean to replace with `object_parts`?

Same bellow with `file_parts` instead.

**Comment by Carreau** [comment]
> This is my first PR. Please let me know if you have any feedback!

BTW, sorry I did only  quick review earlier, without reading first. 

Welcome to GitHub and congratulation on your first PR, you handled git and github well, I know it can be a scary part !

Your PR looks good, as you might see on the UI we have continuous integration that run tests and caught that you actually forgot to rename some of the variables. 

There are several possibility to fix that. 
You can do the modification locally on your own branch and push on GitHub, the PR will update and re-test. 

If you don't have the time, or wish for any reason not to do it, we can do that for you. But we prefer to leave contributor a chance to learn. 

If you have any questions feel free to ask and we;ll do our best to respond.

**Comment by elyashiv** [comment]
Hi.
Notice the delimiter might be different on windows and linux - you should probably use os.sep 
Also, in windows the path can contain a mix of / and \

**Comment by emjobson** [comment]
Hi @Carreau and @elyashiv, thanks for the help and for your patience on my first PR. I will update the code and re-test.

**Comment by Carreau** [comment]
Thanks @emjobson, 
Apologies for the delay in re-reviewing, I did not notice you pushed new commits. 

As a small detail, you may want to update your git config with your email for the commit to be properly attributed to you. 

You can issue `git config --global user.email YOUREMAIL`.

**Comment by Carreau** [comment]
> This is my first PR. Please let me know if you have any feedback!

Congrats on your first PR, and apologies again for the delay in review/merging.

**Pull request merged**